bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -1768,6 +1788,44 @@ public void setDescent (int descent) {
 	}
 }
 
+/**
+ * Forces line heights in receiver to obey provided value. This is
+ * useful with texts that contain glyphs from different scripts,
+ * such as mixing latin glyphs with hieroglyphs or emojis.
+ * <p>
+ * Text lines with different metrics will be forced to fit. This means
+ * painting text in such a way that its baseline is where specified by
+ * given 'metrics'. This can sometimes introduce small visual artifacs,
+ * such as taller lines overpainting or being clipped by content above
+ * and below.
+ * </p>
+ * The possible ways to set FontMetrics include:
+ * <ul>
+ * <li>Obtaining 'FontMetrics' via {@link GC#getFontMetrics}. Note that
+ * this will only obtain metrics for currently selected font and will not
+ * account for font fallbacks (for example, with a latin font selected,
+ * painting hieroglyphs usually involves a fallback font).</li>
+ * <li>Obtaining 'FontMetrics' via a temporary 'TextLayout'. This would
+ * involve setting a desired text sample to 'TextLayout', then measuring
+ * it with {@link org.eclipse.swt.graphics.TextLayout#getLineMetrics(int)}. This approach will also
+ * take fallback fonts into account.</li>
+ * </ul>
+ *
+ * NOTE: Does not currently support (as in, undefined behavior) multi-line
+ * layouts, including those caused by word wrapping. StyledText uses one
+ * TextLayout per line and is only affected by word wrap restriction.
+ *
+ * @since 4.31
```

Comment:
This is wrong version with 99.9% probability (nust be 3.125 as we don't use SDK version but bundle version to track API changes), but unfortunately due a bug in PDE API tooling (https://github.com/eclipse-pde/eclipse.pde/issues/1073) I'm unable to validate API on this PR, and I also assume the API checks are broken on SWT master build either.

Reply:
Thanks for taking you time to have a look!
Yes, sorry, since tags are quite confusing to me. There are 3 different version numbers in SWT (Eclipse, those mentioned in automatic commit messages, and fragment). I'll fix that.

Reply:
done